### ISSUE TITLE
yulInterpreter: datacopy call fails the assertion m_functions.count(_funCall.functionName.name) in ExpressionEvaluator

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#6165]
## Description

(depends on #6180 )

The following code snippet

```
{datacopy(8,1,8)}
```

fails this assertion

https://github.com/ethereum/solidity/blob/6b12726f51c25840b3ccb231018fc37e994869e7/test/tools/yulInterpreter/Interpreter.cpp#L191

## Environment

- Compiler version: develop commit `21e7cadad`
- Target EVM version (as per compiler settings): constantinople
- Framework/IDE (e.g. Truffle or Remix): n/a
- EVM execution environment / backend / blockchain client: n/a
- Operating system: n/a

## Steps to Reproduce

```
$ ./test/tools/yulrun --input-file <file.yul>
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<langutil::InternalCompilerError>'
  what():  
Aborted (core dumped)
```

**Comment by bshastry** [comment]
This is due to object access builtins in the yul interpreter are unimplemented. Temporary solution is to selectively discard test cases containing these builtins.

**Comment by bshastry** [comment]
This is marked for today's release, any ideas how to proceed? Is this a matter of simply avoiding the assertion for certain builtins that are unsupported in the yulInterpreter or adding support for them?

**Comment by bshastry** [comment]
Another scenario that fails the assertion in the title but unrelated to object access functions is the following

```
{
  for{foo() function foo() {}} 0 {}
  {
  }
}
```

My guess is that, while the `for` loop is being interpreted, the interpreter begins to interpret the function call in the init block that is not yet registered.

**Comment by chriseth** [comment]
Oh wow! That is a weird edge case! Can you open a new issue for it? I wonder whether we should disallow function declarations in for init and post blocks....

**Comment by bshastry** [comment]
@chriseth opened a new issue for the for loop init test case (#6532 )
